sig-robotics/proposal/Implement-VSLAM-algorithm-based-on-KubeEdge/README.md

```diff
@@ -0,0 +1,85 @@
+
+The second method offloads all optimizations to the cloud, which greatly reduces the pressure on edge resources, but requires strong network stability. At the same time, the edge side only needs to maintain a small-scale local map (below 10 frames), and the local map can be updated easily by clearing local map on the edge and repalcing it with new received local map update from the cloud.
+
+Based on the above discussion, based on the reliable network communication provided by kubeedge and edgemesh and some open source work available for reference(such as [edge-slam][https://github.com/droneslab/edgeslam]), we intend to use the second method for this work.
+
+### 2.3 General Framework
+
+The general framework is shown in the figure below. The cloud nodes are responsible for most of the operations, and the edge is responsible for data preprocessing, key frame selection, and pose tracking.
+
+![vslam_kubeedge_proposal_arch](images/vslam_kubeedge_proposal_arch.png)
+
+### 2.4 Use Cases
+
+* Users can implement visual SLAM in simulation or real environments by providing ROS-based sensor data topics.
+
+## 3 Design Details
+
+### 3.1 Design Architecture
+
+We will introduce the functions of each module in the form of sub-modules. The overall module design diagram is shown in the figure below. 
+
+![vslam_kubeedge_proposal_design_arch](images/vslam_kubeedge_proposal_design_arch.png)
+
+### 3.2 Tracking Module
+
+Firstly, the edge part contains the Tracking module and the Local map module. The Tracking module performs data preprocessing and local tracking on the sensor data, and determines whether the current frame is a key frame. We use condition 3, 4 and 5 on the Tracking module and use 1, 2 on the cloud. We hope to achieve robust pose tracking while reduce communication usage by select less key-frames. 
+
+If the tracking is loss, the algorithm will recovers pose through global re-localization. The current frame will be transfered to cloud and used to calculate the bag of words, query the recognition database for key-frame candidates.
+
+### 3.3 Map Maintenance
+
+In this work, we will use local map for tracking in the edge and maintain a global map in the cloud. 
+
+**Global Map**
+
+Global map is as same as the map in ORB-SLAM2, which contains the complete set of key-frames, set of map points, the covisibility graph and the essential graph.
+
+**Local Map**
+
+Local map is a subset of the global map, it contains several latest optimized key-frames and map points associated with.
+
+**Map Synchronization**
+
+To maintain the synchronization between local map and global map, the cloud      will send the latest subset of global map to edge frequently. If the update is **redundant** or **latency**, the edge will not use it.
+
+## 4 Workers Communication
```

Comment:
If there is no more information, delete it.

Reply:
Got it!

Reply:
> If there is no more information, delete it.

Got it!